packages/langium/src/lsp/references-provider.ts

```diff
@@ -60,7 +60,7 @@ export class DefaultReferencesProvider implements ReferencesProvider {
         if (targetAstNode) {
             const options = { includeDeclaration: params.context.includeDeclaration };
             this.references.findReferences(targetAstNode, options).forEach(reference => {
-                if (isReference(reference)) {
+                if (isReference(reference) && reference.$refNode) {
```

Comment:
Question: In which cases will a `ReferenceDescription` a proper `Reference`. Is this legacy?

Reply:
Probably yes, I'll remove it.